src/Flysystem/FlysystemAssetStore.php

```diff
@@ -1198,9 +1198,10 @@ function (
         }
 
         // Submit and validate result
-        $result = $callback($fs, $parsedFileID->getFileID());
-        if (!$result) {
-            throw new FlysystemException("Could not save {$filename}");
+        try {
+            $callback($fs, $parsedFileID->getFileID());
+        } catch (UnableToWriteFile $exception) {
+            throw new UnableToWriteFile("Could not save {$filename}");
```

Comment:
```suggestion
        $result = $callback($fs, $parsedFileID->getFileID());
        if (!$result) {
            throw new UnableToWriteFile("Could not save {$filename}");
```

If there's a `UnableToWriteFile` exception thrown from `$callback()` there's no need to catch it only to re-throw a new `UnableToWriteFile` exception

Reply:
Since new filesystem operations no longer return boolean value, if filesystem operation was a success or not. Now they throw `UnableTo*File` exceptions and we should process these exceptions. 
I implemented this changes following this Docs ["No more success result booleans" paragraph](https://flysystem.thephpleague.com/docs/what-is-new/)

Reply:
DONE